### ISSUE TITLE
Plex Web "Dashboard - Now Playing" Not Showing File That's Currently Running in Kodi After PKC 3.6.1 Update

### ISSUE THREAD

**Issue by sting251** [croneter/PlexKodiConnect#1753]
## Initial troubleshooting
I started and stopped numerous shows and movies in Kodi and had intermittent results with them showing up in Plex Web's Now Playing Dashboard. It seemed as though the quicker i would launch a file in Kodi, the lesser chance it would "handshake/resgister" with Plex server via PKC and  show up under Plex Web Now Playing. 

Thinking that perhaps the PKC connection itself to the PMS was being dropped, I added and removed a few files to my monitored Plex libraries WHILE watching a file that was NOT showing up in Plex Now Playing. I found that the updates to my Plex library did in fact sync properly to Kodi via PKC as it should. So that's a good thing! PKC is working and is apparently staying connected to the Plex server even if the file i am watching in Kodi may not be "registering" with Plex "Now Playing" 

## Further troubleshooting:

- Closed Kodi
- Restarted PMS
- Relaunched Kodi and launched a few files to test - Now Playing issue still happening
- Performed full PKC reset, Kodi database reset, etc. Tested some files again - issue still happening
- Reinstalled PKC 3.5.8, reset PKC and Kodi library once again, relaunched Kodi...

Everything from that point onward working perfectly again. Tried about 15 different file launches including several movies and numerous TV episodes. Launched some of them very quickly in Kodi and every single time Plex Web "Now Playing" registered the currently running file with no issues whatsoever. 

So, I turned off Auto-Update on my PKC Add-On and will stick with 3.5.8 for a while. If it ain't broke :) And it def was not broken for me. Kodi 19.3 + PKC 3.5.8 worked flawlessly for me for 2 months and I am on it several hours every day. 

## Steps to reproduce the behavior:

As mentioned above, the Plex Web Now Playing issue is intermittent so the best I can do is describe my particular Plex, Kodi, and PKC environment. 

HTPC: Windows 10 PC
- Latest Plex Media Server 1.25.2.5319-c43dc0277
- Kodi 19.3
- PKC 3.6.1 as of today (was running 3.5.8 for a couple months NO issues at all)
- Direct Paths have always been my method of PKC library connection. Those paths are in place via SMB network drive mappings to my main storage PC (local on my home LAN via CAT6 - everything internal is Gigabit ethernet)

Main Storage PC - Windows 10 
- 16TB Stablebit DrivePool array of four 4TB Seagate IronWolf drives where my shows and movies reside for the Plex library

## Expected behavior
File that is playing in Kodi via PKC, is shown as a currently playing item in Plex Web "Now Playing"

## Two Final Notes

I apologize that my description is lengthy. I am not a software developer or programmer of any kind. I am an IT guy...have been for 20 years but I am not a coder. My brain just doesn't work like you geniuses. I wish it did. I did not even have a GitHub account until just now. I signed up just so i could post this issue. 

PKC is the coolest little piece of 
[kodi.log](https://github.com/croneter/PlexKodiConnect/files/7775455/kodi.log)
[kodi.old.log](https://github.com/croneter/PlexKodiConnect/files/7775456/kodi.old.log)
software I have ever used. I am so thankful for the work done to create such an awesome bridge between Plex and Kodi resulting in the perfect Home Theater solution. I was a Plex guy since day one. However, I always loved Kodi and the limitless customization it allows. As Plex became more mainstream, they gave up on their Plex Home Theater app. PKC to the rescue. So, as a humble 24/7 user and fan of PKC, thank you for all you do and continue to do for folks like myself! 

Merry Christmas/Happy Holidays to all!

**Comment by croneter** [comment]
Thanks very much for all the info - the more the better 😉. If it ain't broken, don't fix it - I'm with you on this one. However, I had to change a lot of stuff around in order to let people correctly see and remotely control different audio and subtitle streams (via Plex Companion). 

Kindly enable debug logging and capture another log file. I can't see anything in your logs, unfortunately. Do wait 5-10 seconds before stopping playback. There's now a wait of at least 5 seconds before playback is reported to the PMS - due to Kodi needing a lot of time to start up / scan the playing streams (and fr... not telling anyone once it's done 😞 ).

**Comment by sting251** [comment]
Will do! Might take me a little while though because Kodi + PKC on the HTPC connected to the 70" Vizio is command central for a huge lineup of shows and movies to watch for the holiday break :) And now that I put the trusty 3.5.8 back in place, there will be some happy, entertained campers in this household. So thanks again for all you do!

**Comment by croneter** [comment]
Glad you signed up to report this bug. Can't test everything myself, unfortunately 😉. Found a related bug while hunting yours, but not the original culprit

**Comment by sting251** [comment]
Okay to my surprise I was able to take a Kodi/PKC entertainment break to watch some NFL. I RDP'd into the HTPC and reinstalled PKC 3.6.1 temporarily for the requested additional testing. I then enabled debug logging in Kodi. 

So yesterday, the Now Playing failing to register issue was occurring a bit more frequently and with multiple files (shows and movies). For this test, I got the issue to manifest 2 out of 11 times using 6 files as test subjects. The Kodi logs attached should show the following:

4:33pm - launched Kodi with PKC 3.6.1 installed:

1. NCIS s09e12 - 4:37pm - worked

2. Power s02e04 - 4:39pm - worked

3. Don't Look Up - 4:41 worked

4. Power s02e04 - 4:42pm - did NOT work

5. NCIS s09e12 - 4:44pm - worked from Resume option

6. Power s02e04 - 4:45pm - worked from Resume option

7. Don't Look Up - 4:46pm - worked from Resume option 

8. Matrix Resurrections - 4:46pm - worked

9. Ancient Aliens - s13e03 4:47pm - worked

10. NCIS s09e12 - 4:47pm - worked

11. Power s02e04 - 4:49pm did NOT work from Resume option

For tests 4 and 11, the Plex Web "Now Playing" did not register the file playing in Kodi. 

Hopefully this is helpful. I just went and reinstalled PKC 3.5.8 so it will definitely be awhile before i can do this again if needed.

Thanks  again!!!

[kodi.log](https://github.com/croneter/PlexKodiConnect/files/7776310/kodi.log)
[kodi.old.log](https://github.com/croneter/PlexKodiConnect/files/7776311/kodi.old.log)

**Comment by croneter** [comment]
Hah, good news, I should have already fixed this with PKC version 2.6.2:
```
DEBUG <general>: PLEX.companion.playqueue: Detected new Kodi element at position 0: {'file': "Z:\\TV Shows\\Power\\Power.S02E04.You're.the.Only.Person.I.Can.Trust.1080p.10bit.BluRay.AAC5.1.HEVC-Vyndros.mkv", 'id': 249, 'label': "04. You're the Only Person I Can Trust", 'title': "You're the Only Person I Can Trust", 'type': 'episode'}
...
ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'AttributeError'>
                                                   Error Contents: 'NoneType' object has no attribute 'startswith'
                                                   Traceback (most recent call last):
                                                     File "C:\Users\WillJ\AppData\Roaming\Kodi\addons\plugin.video.plexkodiconnect\resources\lib\kodimonitor.py", line 67, in onNotification
                                                       self.PlayBackStart(data)
                                                     File "C:\Users\WillJ\AppData\Roaming\Kodi\addons\plugin.video.plexkodiconnect\resources\lib\kodimonitor.py", line 332, in PlayBackStart
                                                       if item.playmethod is None and not path.startswith('plugin://'):
                                                   AttributeError: 'NoneType' object has no attribute 'startswith'
                                                   -->End of Python script error report<--
...
PLEX.companion.playqueue: Detected new Kodi element at position 0: {'file': "Z:\\TV Shows\\Power\\Power.S02E04.You're.the.Only.Person.I.Can.Trust.1080p.10bit.BluRay.AAC5.1.HEVC-Vyndros.mkv", 'id': 249, 'label': "04. You're the Only Person I Can Trust", 'title': "You're the Only Person I Can Trust", 'type': 'episode'} 
...
ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'AttributeError'>
                                                   Error Contents: 'NoneType' object has no attribute 'startswith'
                                                   Traceback (most recent call last):
                                                     File "C:\Users\WillJ\AppData\Roaming\Kodi\addons\plugin.video.plexkodiconnect\resources\lib\kodimonitor.py", line 67, in onNotification
                                                       self.PlayBackStart(data)
                                                     File "C:\Users\WillJ\AppData\Roaming\Kodi\addons\plugin.video.plexkodiconnect\resources\lib\kodimonitor.py", line 332, in PlayBackStart
                                                       if item.playmethod is None and not path.startswith('plugin://'):
                                                   AttributeError: 'NoneType' object has no attribute 'startswith'
                                                   -->End of Python script error report<--
```
